configure.ac

```diff
@@ -71,6 +71,24 @@ case $host in
   ;;
 esac
 
+AC_ARG_WITH([syscall-sandbox],
+  [AS_HELP_STRING([--with-syscall-sandbox=yes|no|kill-without-debug],
```

Comment:
Great feedback. Thanks!

I've now added a `-sandbox=<mode>` parameter.

Available modes are:
* `disallow-and-log` (disallow unexpected syscalls and log the violation, then continue execution)
* `log-and-kill` (log the violation and kill the process)
* `kill` (kill the process immediately in case of a violation (without any logging))

To enable the syscall sandbox `--with-syscall-sandbox` is required at compile time _and_ `-sandbox=<mode>` needs to be specified at run time.

This dual opt-in approach is chosen since this is an experimental feature. We can consider relaxing this in the future when this feature has matured and has been extensively tested in many environments. Note that `--with-syscall-sandbox=no` (no sandbox support) is intentionally the default.

> There still would need to be a --with-seccomp option to be able to build on platforms without BPF, but the implementation and use of --with-seccomp should simpler than this option and not require complicating the build with system/boost process/multiprocess interactions.

Adding `-sandbox=<mode>` allowed for a much simpler `--with-seccomp`, but note that the system/boost process/multiprocess interaction is still there and I'm not sure what the best approach would be if we were to get rid of it:

```
AC_MSG_ERROR(The options --with-syscall-sandbox (which does not allow execve) and --enable-external-signer (which uses execve) are currently incompatible.)
…
AC_MSG_ERROR(The options --with-syscall-sandbox (which does not allow execve) and --enable-multiprocess (which uses execve) are currently incompatible.)
…
AC_DEFINE([HAVE_SYSTEM], [(HAVE_STD__SYSTEM || HAVE_WSYSTEM) && !USE_SYSCALL_SANDBOX], [std::system or ::wsystem, and no syscall sandbox])
```

Isn't say `--with-syscall-sandbox` (no `execve`) and `--enable-external-signer` (yes `execve`) mutually exclusive by design? :) 

Or is the suggestion to detect these features at run-time instead and error if any "`execve` depending" option is passed to `bitcoind` when `-sandbox=<mode>` is used?

Also, note that `--with-syscall-sandbox` will make the functional test be run under the syscall sandbox (using `-sandbox=log-and-kill`). Thus if we allowed for say `--with-syscall-sandbox --enable-external-signer` then all external signer tests would need to be disabled, no?

Please advice :)